From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 6 Aug 2016 11:25:28 +0200
Subject: [PATCH] {NOT UPSTREAMABLE} mm/vmscan: silent -Wunsequenced clang
 warning

clang does not like the way the code is written:

    mm/vmscan.c:2966:25: error: unsequenced modification and access to
    'gfp_mask' [-Werror,-Wunsequenced]
            .gfp_mask = (gfp_mask = memalloc_noio_flags(gfp_mask)),
                                  ^
---
 mm/vmscan.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mm/vmscan.c b/mm/vmscan.c
index 374d95d04178..29120ca14267 100644
--- a/mm/vmscan.c
+++ b/mm/vmscan.c
@@ -2963,8 +2963,6 @@ unsigned long try_to_free_pages(struct zonelist *zonelist, int order,
 	unsigned long nr_reclaimed;
 	struct scan_control sc = {
 		.nr_to_reclaim = SWAP_CLUSTER_MAX,
-		.gfp_mask = (gfp_mask = memalloc_noio_flags(gfp_mask)),
-		.reclaim_idx = gfp_zone(gfp_mask),
 		.order = order,
 		.nodemask = nodemask,
 		.priority = DEF_PRIORITY,
@@ -2972,6 +2970,8 @@ unsigned long try_to_free_pages(struct zonelist *zonelist, int order,
 		.may_unmap = 1,
 		.may_swap = 1,
 	};
+	sc.gfp_mask = (gfp_mask = memalloc_noio_flags(gfp_mask));
+	sc.reclaim_idx = gfp_zone(gfp_mask);
 
 	/*
 	 * Do not enter reclaim if fatal signal was delivered while throttled.
-- 
